### ISSUE TITLE
[Accessibility] No error state or error description in Forms

### ISSUE THREAD

**Issue by rfearing** [primer/brand#556]
### Issue discovered:

In a [resources hub page](https://resources.github.com/devops/fundamentals/devsecops/) form (at the bottom of the page), when the user submits the form without filling in the required fields:

- the `aria-invalid` value remains `false`
- the error message is not displayed
- the error state is not clear

The same behavior can be found [the storybook for the form components](https://stunning-chainsaw-j82glqz.pages.github.io/?path=/story/components-forms-examples--git-hub-enterprise).

#### Steps to reproduce:

1. Go to the [resources hub page](https://resources.github.com/devops/fundamentals/devsecops/), fill out the octocaptcha and submit the form without filling in the required fields.
2. Notice that it (correctly) jumps to the first error field, but the error message is not displayed and the error state styling 0is not different from the normal state.

#### Expected behavior:

When the user submits the form without filling in the required fields, the error message should be displayed, `aria-invalid` should be true and the error state should be clear.

### Accessibility guidelines:
[3.3.1 Error Identification](http://www.w3.org/TR/2008/REC-WCAG20-20081211/#minimize-error-identified): If an [input error](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-identified.html#input-errordef) is automatically detected, the item that is in error is identified and the error is described to the user in text. (Level A)

- Error feedback SHOULD be made available immediately after form submission (or after an equivalent event if there is no form submission event).
- Error feedback **MUST** be programmatically associated with the appropriate element.
- Error feedback **MUST** be programmatically determinable.
- Error feedback **MUST** be meaningful.
- Error feedback **MUST** be visible.

https://github.com/primer/brand/assets/9965014/a45e7529-6337-4929-9bbf-bc44953a5e67

**Comment by rfearing** [comment]
Closed: 

- Validation steps are documented https://primer.style/brand/components/FormControl#validation and I just missed them outright!
